classes/Event/class.xoctEventRenderer.php

```diff
@@ -246,36 +302,90 @@ public function getModalLink(): string
      * @throws ilTemplateException
      * @throws xoctException
      */
-    public function insertDownloadLink(
-        &$tpl,
-        $block_title = 'link',
-        $variable = 'LINK',
-        $button_type = 'btn-info'
-    ): void {
-        if ($download_link_html = $this->getDownloadLinkHTML($button_type)) {
-            $this->insert($tpl, $variable, $download_link_html, $block_title);
+    public function insertDownloadLink(&$tpl, $block_title = 'link', $variable = 'LINK', $button_type = 'btn-info'): void
```

Comment:
The method description is added

Reply:
The reason to pass that &tpl as reference is because, it renders the item as usual like other methods that insert into tpl, when a group contains only one item!